File: mobile/ios/Packages/QervonKit/Sources/QervonSecurity/AppPreferences.swift
// =============================================================================
// File:           mobile/ios/Packages/QervonKit/Sources/QervonSecurity/AppPreferences.swift
// Project:        Qervon
// Author:         USDTG GROUP TECHNOLOGY LLC
// Developer:      Irfan Gedik
// Created Date:   2026-08-12
// Version:        0.1.0
//
// Description:
//   Small, non-sensitive local preferences (UserDefaults-backed). The
//   biometric on/off toggle itself is not a secret; the tokens it gates
//   live in the Keychain (`KeychainTokenStore`).
//
// License:
//   Qervon License v1.0 — see LICENSE in the repository root.
// =============================================================================

import Foundation

public final class AppPreferences: @unchecked Sendable {
    public static let shared = AppPreferences()

    private let defaults: UserDefaults
    private let biometricEnabledKey = "qervon.biometric_unlock_enabled"
    private let lastTenantSlugKey = "qervon.last_tenant_slug"

    public init(defaults: UserDefaults = .standard) {
        self.defaults = defaults
    }

    public var isBiometricUnlockEnabled: Bool {
        get { defaults.bool(forKey: biometricEnabledKey) }
        set { defaults.set(newValue, forKey: biometricEnabledKey) }
    }

    /// Remembered for convenience so the login screen can pre-fill it; a
    /// courier's tenant rarely changes, but the field always stays editable.
    public var lastTenantSlug: String? {
        get { defaults.string(forKey: lastTenantSlugKey) }
        set { defaults.set(newValue, forKey: lastTenantSlugKey) }
    }
}

File: mobile/android/core/security/src/main/kotlin/com/qervon/core/security/AppPreferences.kt
// =============================================================================
// File:           mobile/android/core/security/src/main/kotlin/com/qervon/core/security/AppPreferences.kt
// Project:        Qervon
// Author:         USDTG GROUP TECHNOLOGY LLC
// Developer:      Irfan Gedik
// Created Date:   2026-08-12
// Version:        0.1.0
//
// Description:
//   Plain (non-encrypted) app settings — biometric-lock opt-in, last
//   used tenant slug for convenience on the login screen, etc. Nothing
//   sensitive is stored here; tokens always go through
//   `EncryptedTokenStore`.
//
// License:
//   Qervon License v1.0 — see LICENSE in the repository root.
// =============================================================================

package com.qervon.core.security

import android.content.Context

class AppPreferences(context: Context) {
    private val prefs = context.getSharedPreferences("qervon_app_prefs", Context.MODE_PRIVATE)

    var biometricLockEnabled: Boolean
        get() = prefs.getBoolean(KEY_BIOMETRIC_LOCK, false)
        set(value) = prefs.edit().putBoolean(KEY_BIOMETRIC_LOCK, value).apply()

    var lastTenantSlug: String?
        get() = prefs.getString(KEY_LAST_TENANT_SLUG, null)
        set(value) = prefs.edit().putString(KEY_LAST_TENANT_SLUG, value).apply()

    var courierOnlineOnAppStart: Boolean
        get() = prefs.getBoolean(KEY_COURIER_ONLINE_ON_START, false)
        set(value) = prefs.edit().putBoolean(KEY_COURIER_ONLINE_ON_START, value).apply()

    private companion object {
        const val KEY_BIOMETRIC_LOCK = "biometric_lock_enabled"
        const val KEY_LAST_TENANT_SLUG = "last_tenant_slug"
        const val KEY_COURIER_ONLINE_ON_START = "courier_online_on_start"
    }
}
